pkg/policy/rule.go

```diff
@@ -41,14 +41,16 @@ func (r *rule) validate() error {
 	return nil
 }
 
-func mergeL4Port(ctx *SearchContext, r api.PortRule, p api.PortProtocol, dir string, proto string, resMap L4PolicyMap) int {
+func mergeL4Port(fromEndpoints []api.EndpointSelector, r api.PortRule, p api.PortProtocol,
+	dir string, proto string, resMap L4PolicyMap) int {
+
 	fmt := p.Port + "/" + proto
 	v, ok := resMap[fmt]
 	if !ok {
-		resMap[fmt] = CreateL4Filter(r, p, dir, proto)
+		resMap[fmt] = CreateL4Filter(fromEndpoints, r, p, dir, proto)
 		return 1
 	}
-	l4Filter := CreateL4Filter(r, p, dir, proto)
+	l4Filter := CreateL4Filter(fromEndpoints, r, p, dir, proto)
```

Comment:
We should have a disjunction between rules ("or"). So if rule A is less restrictive than rule B, then rule A should have precedence. But I wouldn't talk about "precedence" because there's no order or priority.